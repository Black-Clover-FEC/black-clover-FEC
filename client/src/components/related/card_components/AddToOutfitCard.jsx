import React from 'react';
import RelatedLib from '../assets/Stylesheet.jsx';
import StyleLib from '../../../assets/Stylesheet.jsx'
import styled from 'styled-components';

const AddToOutfitCard = ({addOutfit}) => {

  return (
    <RelatedLib.card onClick={() => {
      console.log('outfit added')
      addOutfit();
    }}>
      <RelatedLib.plus></RelatedLib.plus>
      <RelatedLib.addOutfitText>
        <StyleLib.h3>Add to Outfit</StyleLib.h3>
      </RelatedLib.addOutfitText>
    </RelatedLib.card>
  )
}

export default AddToOutfitCard;