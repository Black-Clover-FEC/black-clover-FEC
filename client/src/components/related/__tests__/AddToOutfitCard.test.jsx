import React from 'react'; // import like any other jsx file
import renderer from 'react-test-renderer';
import {cleanup, fireEvent, render} from '@testing-library/react';  // used to assert and manipulate rendered components
import AddToOutfitCard from '../card_components/AddToOutfitCard.jsx';

// for more info on testing with jest and react see, https://jestjs.io/docs/tutorial-react

// EXAMPLE TEST CASES

// TEST CHANGES TO COMPONENT RENDERING
it('test to see if the AddToOutfitCard component renders correctly', () => {
  const tree = renderer.create(<AddToOutfitCard />).toJSON();
  expect(tree).toMatchSnapshot();
})