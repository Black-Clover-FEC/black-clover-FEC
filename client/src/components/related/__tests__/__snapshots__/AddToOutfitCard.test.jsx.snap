// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the AddToOutfitCard component renders correctly 1`] = `
<div
  className="sc-cyRfQX dReSUJ"
  onClick={[Function]}
>
  <svg
    className="sc-gjTGSA fOMnxr"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
    />
  </svg>
  <label
    className="sc-fmSAUk sc-eACynP fflqNQ jLnhIs"
  >
    <h3
      className="sc-dmqHEX jScUal"
    >
      Add to Outfit
    </h3>
  </label>
</div>
`;
